core/src/main/java/org/hibernate/ogm/persister/impl/EntityAssociationUpdater.java

```diff
@@ -191,11 +191,13 @@ private AssociationPersister createAssociationPersister(int propertyIndex, Assoc
 				.context();
 
 		Class<?> entityType = persister.getPropertyTypes()[propertyIndex].getReturnedClass();
+		String entityName = persister.getFactory().getClassMetadata( entityType ).getEntityName();
+		OgmEntityPersister entityPersister = (OgmEntityPersister) persister.getFactory().getEntityPersister( entityName );
 
 		AssociationTypeContext associationTypeContext = new AssociationTypeContextImpl(
 				serviceContext.getPropertyOptions( entityType, associationKeyMetadata.getCollectionRole() ),
 				serviceContext.getEntityOptions( entityType ),
-				persister.getTupleTypeContext(),
+				entityPersister.getTupleTypeContext(),
```

Comment:
So as far as I understand it, you pass the persister of the association entity, not the one of the owner? We probably should make it clear that the `persister` field is about the owner of the association.

The field in `AssociationTypeContext` is called `ownerEntityTupleTypeContext` so it should probably be changed. What also bugs me is that `AssociationTypeContextImpl` is instantiated in quite a lot of places and, currently, it's sometimes instantiated with the context of the owner and sometimes with the context of the current association entity.

I'm wondering if we should maybe instantiate it each time with both the current entity context and the owner context but I'm not sure it will not break anything.

But I'm not sure which one should be returned from `AssociationContextImpl#getTupleTypeContext()`...